### PR TITLE
Improved the IDE2 startup and new sketch creation time

### PR DESCRIPTION
**Pull request by kittaakos**
### Reviewer checklist

* [ ] PR addresses a single concern.
* [ ] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-ide/pulls) before creating one)
* [ ] PR title and description are properly filled.
* [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: arduino-ide-extension/package.json [hunk 1]
```diff
@@ -157,10 +158,10 @@
   ],
   "arduino": {
     "cli": {
-      "version": "0.21.0"
+      "version": "20220527"
     },
     "fwuploader": {
-      "version": "2.0.0"
+      "version": "2.2.0"
```
Comment (kittaakos) [inline arduino-ide-extension/package.json:164]: Close https://github.com/arduino/arduino-fwuploader/issues/141 if the error does not happen anymore.

File: arduino-ide-extension/src/browser/arduino-frontend-contribution.tsx [hunk 2]
```diff
@@ -371,6 +342,10 @@ export class ArduinoFrontendContribution
   ): Promise<void> {
     const release = await this.languageServerStartMutex.acquire();
     try {
+      const port = await this.daemon.tryGetPort();
```
Comment (kittaakos) [inline arduino-ide-extension/src/browser/arduino-frontend-contribution.tsx:345]: Move this before `acquire`.

File: arduino-ide-extension/package.json [hunk 3]
```diff
@@ -157,10 +158,10 @@
   ],
   "arduino": {
     "cli": {
-      "version": "0.21.0"
+      "version": "20220527"
```
Comment (fstasi) [inline arduino-ide-extension/package.json:161]:
```suggestion
      "version": "0.23.0"
```

File: arduino-ide-extension/src/browser/arduino-ide-frontend-module.ts [hunk 4]
```diff
@@ -288,8 +290,6 @@ MonacoThemingService.register({
 });
 
 export default new ContainerModule((bind, unbind, isBound, rebind) => {
-  ElementQueries.listen();
-  ElementQueries.init();
```
[the remark this answers is not in the document]
Comment (fstasi) [reply arduino-ide-extension/src/browser/arduino-ide-frontend-module.ts:292]: 🤔 not sure, why do you think is useful? I might be missing something

File: arduino-ide-extension/src/browser/boards/boards-data-menu-updater.ts [hunk 5]
```diff
@@ -31,11 +32,20 @@ export class BoardsDataMenuUpdater implements FrontendApplicationContribution {
...
+  @inject(FrontendApplicationStateService)
+  private readonly appStateService: FrontendApplicationStateService;
+
   protected readonly queue = new PQueue({ autoStart: true, concurrency: 1 });
   protected readonly toDisposeOnBoardChange = new DisposableCollection();
 
   async onStart(): Promise<void> {
-    this.updateMenuActions(this.boardsServiceClient.boardsConfig.selectedBoard);
+    this.appStateService
+      .reachedState('ready')
```
Comment (fstasi) [inline arduino-ide-extension/src/browser/boards/boards-data-menu-updater.ts:43]: minor: I'd probably use enums rather than strings for the reachedStates

**Review by fstasi — APPROVED** [review]

**Pull request merged**